src/libasr/pass/pass_list_concat.cpp

```diff
@@ -0,0 +1,70 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/pass_list_concat.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+#include <utility>
+
+
+namespace LFortran {
+
+using ASR::down_cast;
+using ASR::is_a;
+
+/*
+This ASR pass replaces list-concat to
+
+Converts:
+    x = i + j // lists
+
+to:
+
+ */
+
+
+class ReplaceListConcat : public ASR::BaseExprReplacer<ReplaceListConcat>
+{
+    private:
+
+        Allocator& al;
+
+    public:
+        ReplaceListConcat(Allocator &al_) : al(al_)
+        { }
+
+        void replace_ListConcat(const ASR::ListConcat_t* x) {
+            Location loc = x->base.base.loc;
+            *current_expr = nullptr;
```

Comment:
@czgdp1807 I got questions:
- What is the use of `replace_<node>()`?
- If I place `nullptr` for the `current_expr`, will it remove the `ListConcat` node from ASR?
- Can you please explain the `replacer`?


Does this template look good to start?

Reply:
> What is the use of replace_<node>()?

Well its the method where you have to implement the replacement logic according to the purpose at hand. For example in `replace_ListConcat` you will implement the logic to replace it with the call to the function which is supposed to implement list concatenation.

> * If I place `nullptr` for the `current_expr`, will it remove the `ListConcat` node from ASR?

Well it will replace `ListConcat` with `nullptr`. Instead you should replace it with a `FunctionCall` node.

> Can you please explain the replacer?

Done above.

> How do I verify the asr output? using --show-asr, right?

Only `--show-asr` will give you original ASR. You should also provide `--pass=name_of_the_pass` to get the modified ASR once the pass is applied.

Reply:
I got it, thanks.